client/my-sites/patterns/components/pattern-preview/index.tsx

```diff
@@ -134,7 +164,10 @@ function PatternPreviewFragment( {
 				{ ! canCopy && (
 					<Button
 						className="pattern-preview__get-access"
-						onClick={ () => setIsAuthModalOpen( true ) }
+						onClick={ () => {
+							setIsAuthModalOpen( true );
+							recordOpenModalEvent( 'calypso_pattern_library_get_access' );
```

Comment:
Let's move all tracking events (`opened` / `closed`) inside the modal component, to encapsulate them and keep them in one place.

Reply:
Done. I like this suggestion a lot, thank you!

`recordGetAccessEvent` now handles all four of the events for this modal. the `onOpen` is fired here in the preview, where the button to open the modal sits. The handler is then passed down to the modal component, where it fires the other three events when it receives the event name.

Much less prop passing this way, and fewer new lines overall.